operator/src/main/java/org/keycloak/operator/controllers/WatchedSecretsStore.java

```diff
@@ -185,22 +185,22 @@ private static void cleanObsoleteLabelFromSecret(KubernetesClient client, Secret
                 );
     }
 
-    public static EventSource getWatchedSecretsEventSource(KubernetesClient client, String namespace) {
+    public static EventSource getWatchedSecretsEventSource(KubernetesClient client, Set<String> namespaces) {
         InformerConfiguration<Secret> informerConfiguration = InformerConfiguration
                 .from(Secret.class)
                 .withLabelSelector(Constants.KEYCLOAK_COMPONENT_LABEL + "=" + WATCHED_SECRETS_LABEL_VALUE)
-                .withNamespaces(namespace)
+                .withNamespaces(namespaces)
                 .withSecondaryToPrimaryMapper(secret -> {
                     // get all stores
-                    List<Secret> stores = client.secrets().inNamespace(namespace).withLabel(Constants.COMPONENT_LABEL, COMPONENT).list().getItems();
+                    List<Secret> stores = client.secrets().inNamespace(secret.getMetadata().getNamespace()).withLabel(Constants.COMPONENT_LABEL, COMPONENT).list().getItems();
```

Comment:
What is the difference between `namespaces` `secret.getMetadata().getNamespace()` and `store.getMetadata().getNamespace()` ?

Why did we use only `namespace` before?

Reply:
> What is the difference between namespaces secret.getMetadata().getNamespace() and store.getMetadata().getNamespace() ?

There shouldn't be any the expectation is that both are in the same same namespace.

> Why did we use only namespace before?

There was an assumption that only single namespace mode was supported.